red_black_tree/rb_tree_tests.py

```diff
@@ -0,0 +1,2210 @@
+        self.assertEqual(rb_tree.find_node(7), node_7)
+        self.assertEqual(rb_tree.find_node(6), node_6)
+        self.assertEqual(rb_tree.find_node(9), node_9)
+        self.assertEqual(rb_tree.find_node(15), node_15)
+        # invalid cases
+        self.assertIsNone(rb_tree.find_node(-1))
+        self.assertIsNone(rb_tree.find_node(52454225))
+        self.assertIsNone(rb_tree.find_node(0))
+        self.assertIsNone(rb_tree.find_node(401))
+        self.assertIsNone(rb_tree.find_node(3.00001))
+
+    # ***************TEST INSERTIONS***************
+
+    def test_recoloring_only(self):
+        """
+        Create a red-black tree, add a red node such that we only have to recolor
+        upwards twice
+        add 4, which recolors 2 and 8 to BLACK,
+                6 to RED
+                    -10, 20 to BLACK
+        :return:
+        """
+        tree = RedBlackTree()
+        root = Node(value=10, color=BLACK, parent=None)
+        # LEFT SUBTREE
+        node_m10 = Node(value=-10, color=RED, parent=root) #OK
+        node_6 = Node(value=6, color=BLACK, parent=node_m10) #OK
+        node_8 = Node(value=8, color=RED, parent=node_6, left=NIL_LEAF, right=NIL_LEAF) #OK
+        node_2 = Node(value=2, color=RED, parent=node_6, left=NIL_LEAF, right=NIL_LEAF) #OK
+        node_6.left = node_2 #OK
+        node_6.right = node_8 #OK
+        node_m20 = Node(value=-20, color=BLACK, parent=node_m10, left=NIL_LEAF, right=NIL_LEAF) #OK
+        node_m10.left = node_m20 #OK
+        node_m10.right = node_6 #OK
+
+        # RIGHT SUBTREE
+        node_20 = Node(value=20, color=RED, parent=root) #OK
+        node_15 = Node(value=15, color=BLACK, parent=node_20, left=NIL_LEAF, right=NIL_LEAF) #OK
+        node_25 = Node(25, color=BLACK, parent=node_20, left=NIL_LEAF, right=NIL_LEAF) #OK
+        node_20.left = node_15 #OK
+        node_20.right = node_25 #OK
+
+        root.left = node_m10 #OK
+        root.right = node_20 #OK
+
+        tree.root = root
+        tree.add(4)
+        """
+                    _____10B_____                                     _____10B_____
+               __-10R__        __20R__                           __-10R__        __20R__
+            -20B      6B     15B     25B  --FIRST RECOLOR-->  -20B      6R     15B     25B
+                    2R  8R                                            2B  8B
+               Add-->4R                                                4R
+
+
+
+                                  _____10B_____
+                             __-10B__        __20B__
+   --SECOND RECOLOR-->    -20B      6R     15B     25B
+                                  2B  8B
+                                   4R
+        """
+        """ This should trigger two recolors.
+            2 and 8 should turn to black,
+            6 should turn to red,
+            -10 and 20 should turn to black
+            10 should try to turn to red, but since it's the root it can't be black"""
```

Comment:
@Enether You might notice that Codacy is not giving errors on all of your triple quotes comments. This one got a flag because you used two comment blocks in succession. I don't see a reason for that.